### ISSUE TITLE
Unreproducible results when directly import keras in CPU environment

### ISSUE THREAD

**Issue by nemo9cby** [keras-team/keras#14671]
I have met reproducibility issues regarding the keras version 2.2.2, 2.2.3, 2.2.4, and 2.2.5 when training on the CPU environment with tensorflow version 1.14.0. The environment is built through conda. One of the environment is shown below (where I only do pip install keras==2.2.4 and pip install tensorflow ==1.14.0).
The python version is 3.6, created by conda create -n keras224 python=3.6. The conda version is 4.9.2. 
The OS I am using is Ubuntu 18.04.
![image](https://user-images.githubusercontent.com/1452417/119698777-951e8100-be1f-11eb-9af7-4bccd686b740.png)

The code I am using is shown in (https://github.com/nemo9cby/repro_exp/blob/main/LeNet_keras.py) and https://github.com/nemo9cby/repro_exp/blob/main/LeNet_keras_first.py. The two files were modified from https://github.com/BIGBALLON/cifar-10-cnn/blob/master/1_Lecun_Network/LeNet_keras.py 
For demonstration purposes, each training is configured to last for 2 epochs.
The differences between these two files are the sequences of import statements. The first file LeNet_keras.py import tensorflow first as shown below:
![image](https://user-images.githubusercontent.com/1452417/119698881-aebfc880-be1f-11eb-8f86-da6d034a3be4.png)
The second file LeNet_keras_first.py import keras first as shown below:
![image](https://user-images.githubusercontent.com/1452417/119698909-b5e6d680-be1f-11eb-91a2-c595b533c9ed.png)

When running, I have applied the best practices of controlling the randomness in the training scripts described here
https://stackoverflow.com/questions/48631576/reproducible-results-using-keras-with-tensorflow-backend, which are:
1.	Setting pythonhashseed 
2.	Setting python random seed;
3.	Setting numpy random seed;
4.	Setting tensorflow random seed; and 
5.	Setting the intra_ops_parallelism_threads and inter_ops_parallelism_threads  to 1 to control the number of threads.

When we import keras first, the results cannot be exactly reproducible by running the scripts. For example,
First time running results:
![image](https://user-images.githubusercontent.com/1452417/119698984-c7c87980-be1f-11eb-92a8-0817e271a7b1.png)

Second time running results:
![image](https://user-images.githubusercontent.com/1452417/119699101-ce56f100-be1f-11eb-8d16-d562b6baad02.png)

However, when we change all import statements from `import keras.xx` to `import tensorflow.keras.xx`, the results can be exactly reproducible even without setting the number of threads.
When we ran multiple times, it always returns the identical results shown below:
![image](https://user-images.githubusercontent.com/1452417/119699167-dca50d00-be1f-11eb-938a-7dd822f7f228.png)

**Comment by jvishnuvardhan** [comment]
@nemo9cby Can you please try with recent TF/keras versions? I tried with `TF2.6` and `python 3.8` and I didn't notice any reproducibility issues.  Thanks!

**Comment by nemo9cby** [comment]
Hi @jvishnuvardhan, thanks for your reply. Yes, as mentioned in our description, I have observed this issue in Keras versions ranging from 2.2.2 and 2.2.5. I am wondering if any solutions exist for the old versions of Keras?

**Comment by jvishnuvardhan** [comment]
@nemo9cby There are lot of changes between keras 2.2.2 to 2.6. With the recent changes, going forward, we are supporting `TF2.6` or above. I would suggest you to try recent TF/keras version and let us know if this issue persists. For older issue, even if we find that there is an issue, we are not going to cherry pick update older versions. Thanks!
